### PR TITLE
feat(KIT-0104): bootstrap becomes an exec shim; agentive new --no-kit (PR 2 of 3)

### PR DESCRIPTION
**Pull request by movito**
## Summary

PR 2 of 3 for KIT-0104 (KIT-ADR-0030 — the door ships in the package). PR 1 (#129) shipped `agentive new` / `agentive adopt`; this PR retires the old entrance:

- **F3 — `scripts/local/bootstrap` is an exec shim** (1,230 → 165 lines): translates `--new`/`--adopt` to the package verbs and execs the **checkout's own** `packages/agentive-kit/src` (deterministic — never an older installed wheel). `--help` defers to the package. One legacy branch stays: `--adopt --design-materials` still drives `engine-materials.sh` directly until project-intake (KIT-0105) succeeds it — the branch dies with the shim.
- **F4 — `agentive new <dir> --no-kit`** produces a KIT-ADR-0032 rung-0 repo (check hook + git init; no `.kit/`, no record, no doctor), mirroring the adopt rung-0 tail: explicit offers and a preset `env-source` are acknowledged out loud; `--bots` and `--name`/`--prefix` are refused loudly (masking class).
- **F2 riders filed and pinned**: shim removal = `.kit/tasks/1-backlog/KIT-0107-remove-the-bootstrap-shim.md`; engine consolidation = `.kit/tasks/1-backlog/KIT-0108-consolidate-the-door-engines.md` — both pinned to the next minor release (0.10.0).

Two deliberate retirements ride the shim (both operator decisions recorded on PR 1):
- **adopt is packaged-mode** — the legacy copy-adopt (toolchain rsync) is gone; pinned by `TestPackagedAdoptThroughShim`.
- **`--no-kit` is rung 0 from both verbs** — the old seeded-record `--no-kit` is gone; pinned by `TestRungZeroThroughShim`.

## Grep proofs

**Matrix single ownership** (AC 5) — the only implementation is the Python front; also pinned forever as `tests/test_setup_door.py::TestShimStatic`:
```text
$ grep -rn 'LEGAL_PAIRS\|single:python\|legal_pairs' --include='*.sh' --include='*.py' --include=bootstrap scripts/ packages/ .claude/ | grep -v door/__init__.py
(no output)
```

**Behavioral equivalence** (AC \"bootstrap --new proof\") — `TestShimEquivalence` asserts tree-snapshot equality (sha256, per file) between a shim run and a direct `agentive` run for both `new` and `adopt`; the full scaffold-acceptance suite also runs through the shim.

## Deliberate deviations (said out loud)

- The shim does not re-implement the old TTY mode prompt: bare `bootstrap` exits 2 with `mode is required` in all contexts (the flag surface is unchanged).
- `--design-materials` accepts no companion flags (previously-legal redundant ones like `--shape single` are now refused loudly rather than resolved) — the flow is fixed adopt/single/python, and the branch is deprecated.

## Review-surface budget

832 additions / 3,263 deletions. Over the 500-addition target — **operator-approved as one PR** (in-session, 2026-08-13): 64% of additions are test rewrites replacing 3.2k deleted legacy-characterization lines, 17% are the two required backlog filings + CHANGELOG; the shim breaks the legacy tests instantly, so no green split lands under budget.

## Evaluator trio (pre-PR, KIT-0035 ordering)

Record: `.kit/context/reviews/KIT-0104-evaluator-review.md` (PR 2 section). fast = CONCERNS (3 parity notes, declined with evidence); deep = FAIL (both \"blockers\" refuted against `main`'s door — the bash door never accepted dash-leading targets and its identity check/GIT_* scrub are byte-parity, the KIT-0048 defense); claude-code = APPROVED (one remediation taken: the shim probes `agentive_kit/cli.py`, not just the package dir).

## Test plan

- [x] Full suite: 1110 passed, 13 skipped (`ci-check.sh` green, coverage 89.6%)
- [x] `test_door_data_sync.py` untouched and green (no engine/data edits in this PR)
- [x] Shim smoke: `--new` full install, `--new --no-kit` rung 0, `--help` deferral, materials-flag refusals
- [x] New: `TestNewNoKit` (5 E2E), `validate_combo` rung-0 units, shim static/translation/materials suites

Task: `.kit/tasks/3-in-progress/KIT-0104-ship-the-door-in-the-package.md` (stays in-progress; PR 3 = prose sweep + KIT-0094 rider)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

https://claude.ai/code/session_01FvXHQsb72XdAs8wqzQYuoQ

---

> [!NOTE]
> **Medium Risk**
> Changes the primary project-creation entry path and rung-0 semantics for `new`, with large test rewrites; behavior is heavily pinned but mis-shimmed flags or materials-only edge cases could still surprise operators until KIT-0107 lands.
> 
> **Overview**
> **KIT-0104 PR 2** turns `scripts/local/bootstrap` from the full door (~1,230 lines) into a **~165-line exec shim** that maps `--new` / `--adopt` to `python3 -m agentive_kit.cli` using the **checkout’s** `packages/agentive-kit/src` (not an installed wheel). Matrix, presets, help, and exit codes live only in `agentive_kit.door`. The shim keeps **one legacy path**: `--adopt --design-materials` still runs `engine-materials.sh` until KIT-0107; everything else prints a removal notice (KIT-0107, 0.10.0).
> 
> In the package, **`--no-kit` is rung 0 for both verbs** (KIT-0104 F4 / KIT-ADR-0032): check hook + git init, no `.kit/`, record, `.env`, or doctor. New guards refuse **`--no-kit` with `--name`/`--prefix` or `--bots`**, and preset **`env-source` on rung 0** is acknowledged but not applied. Help text matches (`.env` only when a kit install runs).
> 
> **Tests** drop legacy copy-adopt characterization and add **shim vs direct `agentive` tree equivalence**, rung-0 coverage (`TestNewNoKit`, shim rung-zero suites), and **bots conformance** via `pkg_door.load_record` / `normalize_bots`. **Backlog**: KIT-0107 (shim + materials branch removal) and KIT-0108 (engine consolidation), both pinned to **0.10.0**.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 7971aef801a65ccb4ba061df98ed8bf68c37ec03. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: scripts/local/bootstrap [hunk 1]
```diff
@@ -1,1230 +1,167 @@
...
-    # whose record says profile: none (BugBot round 2). Ignoring the
-    # answer is said out loud — never a silent drop.
-    if [ "$PROFILE" = "python" ] && [ -z "$WITH_VENV" ] && offer="$(preset_get venv)"; then
-        # validate BEFORE deciding applicability — a malformed value
-        # fails loud even when the answer would be ignored (N2)
-        case "$offer" in
-            yes|no) ;;
-            *) die_usage "preset key 'venv' must be yes or no (got: $offer)" ;;
+    [ -d "$TARGET" ] ||
+        die_usage "--adopt target does not exist: $TARGET"
```
[the remark this answers is not in the document]
Comment (movito) [reply scripts/local/bootstrap:137]: Fixed in 7971aef: the materials branch now absolutizes the target and refuses the kit checkout itself ("target is the kit source repo itself"), restoring the old door's guard that this branch bypassed by not reaching the package. Pinned by TestMaterialsBranch::test_kit_checkout_target_refused. Good catch — this was the one real behavioral gap in the shim.

File: .kit/tasks/1-backlog/KIT-0107-remove-the-bootstrap-shim.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  resurrected copy hard-fails.
+- **F2 — the materials flow retires with it.** By then the
+  `project-intake` agent (KIT-0105, blocked on KIT-0104) is the
+  materials successor. Remove `engine-materials.sh` + its tests
+  (`tests/test_engine_materials.py`, the materials E2E in
+  `test_entrance_shims.py`) or, if KIT-0105 has not landed, STOP and
+  re-sequence — the kit must not lose its only materials path.
+- **F3 — the packaged door's `--design-materials` refusal drops its
+  "while the shim lasts" pointer** (`agentive_kit/door/__init__.py`)
+  and points only at project-intake.
```
[the remark this answers is not in the document]
Comment (movito) [reply .kit/tasks/1-backlog/KIT-0107-remove-the-bootstrap-shim.md]: Fixed in 7971aef: both references are repository-relative now (packages/agentive-kit/src/... and tests/ prefixes).

File: .kit/tasks/1-backlog/KIT-0108-consolidate-the-door-engines.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+- **F1 — one home per engine.** Either port the two engines' behavior
+  to Python inside `agentive_kit.door` (preferred — ends the bash 3.2 /
+  BSD-userland constraint for good) or make the packaged copies the
+  ONLY copies with the kit tree consuming them from the package. No
+  byte-pinned twins either way.
+- **F2 — the faux-kit-root staging disappears** once no engine resolves
+  `SCRIPT_DIR/../..` (`stage_door_root` and `_STAGE_MAP` shrink or go).
+- **F3 — `tests/test_door_data_sync.py` retires** with the duplication;
+  its "no extra files" direction survives wherever a data store remains.
```
[the remark this answers is not in the document]
Comment (movito) [reply .kit/tasks/1-backlog/KIT-0108-consolidate-the-door-engines.md:38]: Fixed in 7971aef: the acceptance criterion now requires an EXECUTABLE one-source-path-per-engine assertion that fails while both copies exist; the grep proof is demoted to a supplement.

File: packages/agentive-kit/src/agentive_kit/door/__init__.py [hunk 4]
```diff
@@ -656,7 +661,7 @@ def usage_text(mode: str) -> str:
   --target-path / --target-github
                        product-repo pointer (planning only)
   --no-kit             rung 0 (KIT-ADR-0032): plain repo, no .kit/, no
-                       kit install (adopt, single shape only)
+                       kit install (single shape; both verbs)
```
[the remark this answers is not in the document]
Comment (movito) [reply packages/agentive-kit/src/agentive_kit/door/__init__.py:664]: Fixed in 7971aef with your proposed wording (adapted): the guarantee is scoped to --new targets WITH a kit install, and the rung-0 exception is stated in the same paragraph.

File: scripts/local/bootstrap [hunk 5]
```diff
@@ -1,1230 +1,167 @@
...
-        echo "Install the agent plugin:"
-        echo "    claude plugin marketplace add movito/agentive-skills"
-        echo "    claude plugin install agentive-workflow@agentive-skills"
-        if ! command -v claude >/dev/null 2>&1; then
-            echo "  (needs the Claude Code CLI on PATH first)"
-        fi
-    fi
-}
+    if [ $# -gt 0 ]; then shift; fi
+done
```
[the remark this answers is not in the document]
Comment (movito) [reply scripts/local/bootstrap:119]: Fixed in 7971aef with your set_mode approach: a second mode flag exits 2 ("only one mode flag is allowed") instead of silently dropping the first target. Worth noting the old door had the same last-wins behavior — but the masking-class rule is the right standard, so the shim is now stricter-loud. Pinned by test_second_mode_flag_refused_never_drops_first_target.

File: scripts/local/bootstrap [hunk 6]
```diff
@@ -1,1230 +1,167 @@
...
-    if [ "$MODE" = "new" ]; then
-        if [ -n "$ENV_SOURCE" ]; then
-            apply_env_source "$ENV_SOURCE"
-        else
-            seed_env_from_template
-            offer_env_carryover
-        fi
-        fill_env_identity
-    fi
+    done
```
[the remark this answers is not in the document]
Comment (movito) [reply scripts/local/bootstrap:160]: Fixed in 7971aef, taking your second option: the sweep keeps rejecting --no-preset (the materials engine reads no preset, so accepting the flag would be a silent no-op), the docstring now says so explicitly, and a --no-preset rejection row is added to test_other_flags_refused_never_dropped.

File: tests/agentive_kit/test_door_e2e.py [hunk 7]
```diff
@@ -264,6 +264,110 @@ def test_no_python_toolchain(self, new_planning):
...
+            text=True,
+            timeout=30,
+        )
+        assert branch.stdout.strip() == "main"
+        log = subprocess.run(
+            ["git", "-C", str(target), "log", "--oneline"],
+            capture_output=True,
+            text=True,
+            timeout=30,
+        )
```
[the remark this answers is not in the document]
Comment (movito) [reply tests/agentive_kit/test_door_e2e.py]: Fixed in 7971aef: both git probes in TestNewNoKit now pass env=env, matching the door run and the test_bootstrap_shapes precedent. (The session-scoped conftest GIT_* scrub already covers the suite, so this is belt-and-braces — but consistency wins.)

File: scripts/local/bootstrap [hunk 8]
```diff
@@ -1,1230 +1,183 @@
...
+            TARGET="${1:-}"
+            ;;
+        --new=*)   set_mode new; TARGET="${1#--new=}" ;;
+        --adopt)
+            set_mode adopt
+            shift
+            reject_flaglike --adopt "${1:-}"
+            TARGET="${1:-}"
+            ;;
+        --adopt=*) set_mode adopt; TARGET="${1#--adopt=}" ;;
```
[the remark this answers is not in the document]
Comment (movito) [reply scripts/local/bootstrap]: Fixed in 052ccac with your proposed diff: both equals-form arms now run reject_flaglike on the extracted suffix, so --new=--help is a usage error (exit 2), matching the split-argument arms. Regression rows added for --new=--help and --adopt=-h.
